File: src/2125_number_of_laser_beams_in_a_bank/number_of_laser_beams_in_a_bank.cpp
#include <iostream>
#include <string>
#include <vector>
using namespace std;

class Solution {
   public:
    int numberOfBeams(vector<string>& bank) {
        int prev = 0;
        int result = 0;

        for (string lasers : bank) {
            int cur = 0;

            for (char c : lasers) {
                if (c == '1') {
                    cur++;
                }
            }
            if (cur == 0) continue;

            result += cur * prev;
            prev = cur;
        }

        return result;
    }
};

int main() {
    Solution s;

    vector<string> bank1 = {"011001", "000000", "010100", "001000"};
    cout << s.numberOfBeams(bank1) << " Expected: 8" << endl;

    vector<string> bank2 = {"000", "111", "000"};
    cout << s.numberOfBeams(bank2) << " Expected: 0" << endl;

    return 0;
}

File: src/2125_number_of_laser_beams_in_a_bank/number_of_laser_beams_in_a_bank.py
from typing import List


class Solution:
    def numberOfBeams(self, bank: List[str]) -> int:
        previous_count = 0
        result = 0

        for lasers in bank:
            count = lasers.count("1")
            if count > 0:
                if previous_count > 0:
                    result += previous_count * count

                previous_count = count

        return result


s = Solution()

bank = ["011001", "000000", "010100", "001000"]
print(s.numberOfBeams(bank), 8)

bank = ["000", "111", "000"]
print(s.numberOfBeams(bank), 0)
